### ISSUE TITLE
C# 9's target-typed new doesn't work with System.Runtime.CompilerServices.CallerMemberName attribute

### ISSUE THREAD

**Issue by Edward-Alchikha** [dotnet/roslyn#50475]
**Version Used**: 
.NET 5.0.102 on Ubuntu 20.04

**Steps to Reproduce**:

```csharp
using System;
using System.Runtime.CompilerServices;

namespace ConsoleApp1
{
    class Test
    {
        public string Name { get; }
        
        public Test([CallerMemberName] string name = "")
        {
            Name = name;
        }
    }

    class Program
    {
        private static Test test1 = new Test();
        private static Test test2 = new();
        
        static void Main()
        {
            Console.WriteLine($"1: {test1.Name}, 2: {test2.Name}");
        }
    }
}
```

**Expected Behavior**:
It should output "1: test1, 2: test2"

**Actual Behavior**:
It actually outputs "1: test1, 2: "

**Comment by Youssef1313** [comment]
https://github.com/dotnet/roslyn/issues/49547

**Comment by HobbsCode** [comment]
The problem was recently addressed. The code in question performs correctly when running against master on [sharplab](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAgrgOwD4AEBMAjALABQuADAAS6EB0ASljAJZhx0DCA9mAA4sANogDKiAG4sAxnCgBuMmQJUAKrJhkA3mSq6aAZhqFqAOQjsqmqgHM4MOVQC+i0noOr1ACgDanCIOEEAFk4MGBEM3YAXSNqTHM4KgBeKgAiVIBKHT1tVzc9SMSU+PYFPN1nUkqlfCMAdi1s3T4EFnEIeCMANg9YKnhYQmSqTDgAd16YTwyytxa2jsTaHrU+gZha4vHpsqbumgAWKiCIFkxpvdz8vVoATk8AElTCEEt1+kLHJCp8V811/B0T6ZWZ6SqOIA=).

**Comment by Edward-Alchikha** [comment]
Looking at https://github.com/dotnet/roslyn/issues/49547, it looks like this has been fixed in master since at least November 22, but in .NET 5.0.102 (released January 12) it's not fixed. When will this fix make it to a release?

**Comment by jaredpar** [comment]
This will be released with VS 16.9 and it's accompanying .NET SDK release (believe that will be the 5.0.200 series).
